File: Math and Logic/Reverse Number/ReverseNumber.py
def reverseAlternateSolution(number):
    return int(str(number)[::-1])

def reverse(number):
    
    reversedNumber = 0

    while number > 0:

        reversedNumber = (reversedNumber * 10) + (number % 10)
        number = number // 10

    return reversedNumber

number = int(input("Enter a number: ").strip())
print("Reversed number: ", reverse(number))

File: Math and Logic/Reverse Number/ReverseNumber.cpp
#include <iostream>

using namespace std;

int reverse(int number)
{

    int reversedNumber = 0;

    while (number > 0)
    {
        reversedNumber = (reversedNumber * 10) + (number % 10);
        number = (int)(number / 10);
    }

    return reversedNumber;
}

int main()
{

    int number;

    cout << "Enter a number: ";
    cin >> number;

    cout << "Reversed number: " << reverse(number);
    return 0;
}